services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/share_activity_pack/shareActivityPackModal.tsx

```diff
@@ -1,43 +1,102 @@
   const [classroomUnits, setClassroomUnits] = React.useState<ClassroomUnit[]>([]);
   const [link, setLink] = React.useState<string>(classrooms && classrooms.length && getDefaultLink());
 
   React.useEffect(() => {
-    if(!classrooms.length || (!classroomUnits.length && unitId)) {
+    if(!classroomUnits.length && unitId) {
       requestGet(`/teachers/classrooms/classrooms_and_classroom_units_for_activity_share/${unitId}`, (body) => {
-        setClassrooms(body.classrooms && body.classrooms.classrooms_and_their_students)
+        const classrooms = body.classrooms && filterClassrooms(body.classrooms.classrooms_and_their_students)
+        setClassrooms(classrooms)
         setClassroomUnits(body.classroom_units)
       })
     }
-  }, [])
+  }, []);
 
-  function getDefaultLink() {
-    if(classrooms.length !== 1) {
-      return ''
+  React.useEffect(() => {
+    if(classrooms && classrooms.length) {
+      const newLink = getDefaultLink();
+      const filteredClassrooms = filterClassrooms(classrooms);
+      getSelectedClass(filteredClassrooms);
+      newLink && setLink(newLink);
     }
-    const classroomObject = classrooms[0];
-    const classroomUnitObject = classroomUnits[0];
-    const { classroom } = classroomObject;
-    if(!singleActivity) {
-      const { id } = classroom;
-      return `${process.env.DEFAULT_URL}/classrooms/${id}?unit_id=${unitId}`
-    } else {
-      const { id } = classroomUnitObject;
-      return `${process.env.DEFAULT_URL}/classroom_units/${id}/activities/${singleActivity.id}`;
+  }, [classrooms]);
+
+  React.useEffect(() => {
+    if(classrooms && classrooms.length) {
+      const filteredClassrooms = filterClassrooms(classrooms);
+      setClassrooms(filteredClassrooms);
     }
+  }, [selectedClassroomId, selectedClassroomName, selectableClassrooms]);
+
+  function filterClassrooms(classrooms) {
+    if(selectedClassroomName) {
+      return classrooms.filter(classroomObject => classroomObject.classroom.name === selectedClassroomName);
+    } else if(selectedClassroomId === ALL_CLASSES) {
+      return classrooms.filter(classroomObject => selectableClassrooms.includes(classroomObject.classroom.name))
+    } else if(selectedClassroomId) {
+      const id = parseInt(selectedClassroomId);
+      return classrooms.filter(classroomObject => classroomObject.classroom.id === id);
+    }
+    return classrooms;
+  }
+
+  function getSelectedClass(filteredClassrooms) {
+    // we want set selectedClass and return only the link without dropdown if only one class remains after filtering
+    if(filteredClassrooms && filteredClassrooms.length === 1) {
+      const classroomObject = filteredClassrooms[0];
+      const { classroom } = classroomObject;
+      const { id, name } = classroom;
+      const dropdownOption = {
+        value: id,
+        label: name
+      }
+      setSelectedClass(dropdownOption);
+    }
+  }
+
+  function getDefaultLink() {
+    if(classrooms && classrooms.length) {
```

Comment:
if you just do a `if (!(classrooms && classrooms.length)) { return '' }` here you can avoid the nested conditionals.